go/vt/vtgate/planbuilder/ddl.go

```diff
@@ -150,6 +154,30 @@ func buildDDLPlans(ctx context.Context, sql string, ddlStatement sqlparser.DDLSt
 		}, nil
 }
 
+func buildDropProcedurePlan(vschema plancontext.VSchema, dp *sqlparser.DropProcedure) (key.ShardDestination, *vindexes.Keyspace, error) {
+	destination, keyspace, _, err := vschema.TargetDestination(dp.Name.Qualifier.String())
+	if err != nil {
+		return nil, nil, err
+	}
+	// Clear out the qualifier from the table name.
+	dp.SetTable("", dp.Name.Name.String())
+	return destination, keyspace, nil
+}
+
+func buildCreateProcedurePlan(vschema plancontext.VSchema, cp *sqlparser.CreateProcedure) (key.ShardDestination, *vindexes.Keyspace, error) {
+	destination, keyspace, _, err := vschema.TargetDestination(cp.Name.Qualifier.String())
+	if err != nil {
+		return nil, nil, err
```

Comment:
verify if this works when stored procedure does not have a qualifier and there is a single keyspace. It get's routed to it.

Reply:
Added.